### PR TITLE
Bugfix/pro 4187/secondary ecu paths

### PR DESCRIPTION
**Pull request by patrickvacek**
This is really just a pretty basic path handling fixup, but along the way I fixed up a bunch of other stuff. This was one of those bugs where I think I had to learn everything the hard way.

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -64,7 +78,6 @@ function(add_aktualizr_test)
                      ${WD})
         endif()
     add_dependencies(build_tests t_${AKTUALIZR_TEST_NAME})
-    aktualizr_source_file_checks(${AKTUALIZR_TEST_SOURCES})
```
Comment (cajun-rat) [inline tests/CMakeLists.txt]: What about adding an option to `add_aktualizr_test` to skip adding the sources?  That would remove one more place for people to forget. Adding the .h files is a good catch, though

Comment (patrickvacek) [reply tests/CMakeLists.txt]: Possible, but this gets tricky fast. The problematic file is fake_ostree.cc. It is used as a supplemental file by two tests, both of which also have an individual .cc as well. Hence, if you skipped adding the sources of one, you'd still have to manually add the individual .cc.

But I was sad to have to break your logic there, so maybe that is worth it. (Or is there a way to skip duplicates? Or could `add_aktualizr_test` maybe just add the files to a list, which can be de-dup'd, and then run `aktualizr_source_file_checks` on the list at the end?)

**Comment by patrickvacek** [comment]
I managed to fix the issue with `aktualizr_source_file_checks` about as ideally as I think is possible. Files are automatically added to a list in `add_aktualizr_test` and can be automatically added as well, and the list is de-duped to prevent ugly CMake errors before handing off the list to `aktualizr_source_file_checks`.

File: src/uptane/secondaryconfig.h [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
-    boost::filesystem::path firmware_path;
+namespace Uptane {
+class SecondaryConfig {
+ public:
+  std::string ecu_private_key() const;
+  std::string ecu_public_key() const;
+
+  std::string ecu_serial;
+  std::string ecu_hardware_id;
+  std::string ecu_private_key_;
```
Comment (patriotyk) [inline src/uptane/secondaryconfig.h]: ecu_private_key_ and ecu_public_key_ should be private?

Comment (patrickvacek) [reply src/uptane/secondaryconfig.h]: I have wondered about that. For now, I'm following the convention we set with the TlsConfig, which is that you can get/set the strings directly, but if you want to get the path with the directory prefixed automatically, you have to use the function. We could perhaps use a different name for the function if that would help make it clearer.

File: src/uptane/secondary.cc [hunk 3]
```diff
@@ -59,18 +58,24 @@ Json::Value Secondary::newTargetsCallBack(const std::vector<Uptane::Target> &tar
 }
 
 void Secondary::setKeys(const std::string &public_key, const std::string &private_key) {
-  if (!boost::filesystem::exists(config.full_client_dir)) {
-    boost::filesystem::create_directories(config.full_client_dir);
+  boost::filesystem::path private_parent_dir = boost::filesystem::path(config.ecu_private_key()).parent_path();
+  if (!private_parent_dir.empty()) {
+    boost::filesystem::create_directories(private_parent_dir);
```
Comment (cajun-rat) [inline src/uptane/secondary.cc]: We might want to think about adding an optional enum second parameter to Utils::writeFile to optionally create any necessary parent directories.

Comment (patrickvacek) [reply src/uptane/secondary.cc]: I am in favor of that!

**Pull request merged**